src/spillway_srv.erl

```diff
@@ -65,7 +64,8 @@ enter(Name, Size, Limit) when Size > 0 ->
 
             [OldValue, NewValue] = ets:update_counter(?TID, Name,
                 [{#counter.value, 0},
-                    {#counter.value, Size, Limit, Limit}]),
+                    {#counter.value, Size, Limit, Limit}],
+                     #counter{name = Name}),
```

Comment:
Function comment should be updated to take this dynamic counter initialization into account.